### PR TITLE
draksetup: Add snapshot export/import

### PR DESCRIPTION
**Pull request by chivay**
~~TODO: Spawn a VM after importing minimal snapshot (to do postinstall)~~

### CODE REVIEW

**Comment by icedevml** [comment]
looks very cool overall :+1:

**Comment by icedevml** [comment]
@chivay This was related to the CI as well

File: drakrun/drakrun/draksetup.py [hunk 1]
```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
     subprocess.run(['xl', 'qemu-monitor-command', domain_name, f'change ide-5632 {iso_path_full}'])
 
 
+def get_minio_client(config):
+    minio_cfg = config['minio']
+    mc = Minio(endpoint=minio_cfg['address'],
+               access_key=minio_cfg['access_key'],
+               secret_key=minio_cfg['secret_key'],
+               secure=minio_cfg.getboolean('secure', fallback=True))
+    return mc
```
Comment (BonusPlay) [inline drakrun/drakrun/draksetup.py:491]:
```suggestion
    return Minio(endpoint=minio_cfg['address'],
               access_key=minio_cfg['access_key'],
               secret_key=minio_cfg['secret_key'],
               secure=minio_cfg.getboolean('secure', fallback=True))
```

File: drakrun/drakrun/draksetup.py [hunk 2]
```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
...
+
+
+def minio_prefix_empty(mc, bucket, prefix):
+    return len(list(mc.list_objects(bucket, prefix=prefix))) == 0
```
Comment (BonusPlay) [inline drakrun/drakrun/draksetup.py:495]: This is used referenced only one in code and the logic is quite simple. Should this be a separate function?

File: drakrun/drakrun/draksetup.py [hunk 3]
```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
...
+    logging.info("Downloading VM config")
+    mc.fget_object(bucket, f"{name}/cfg.template", os.path.join(ETC_DIR, "scripts", "cfg.template"))
+
+    # Now we have imported InstallInfo object
+    install_info = InstallInfo.load()
+
+    # Patch ZFS pool name
+    if zpool is not None:
+        install_info.zfs_tank_name = zpool
+        install_info.save()
```
Comment (BonusPlay) [inline drakrun/drakrun/draksetup.py:599]: Why do we save the info here?

File: drakrun/drakrun/draksetup.py [hunk 4]
```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
...
+
+    with tempfile.NamedTemporaryFile() as disk_image:
+        logging.info("Downloading VM disk image")
+        mc.fget_object(bucket, f"{name}/disk.img", disk_image.name)
+
+        logging.info("Importing VM disk")
+        storage.import_vm0(disk_image.name)
+
+    # This could probably use some refactoring
+    # We're duplicating quite a lot of code from install function
```
Comment (BonusPlay) [inline drakrun/drakrun/draksetup.py:611]: I agree, we should split `draksetup.py` into multiple files while we're at it.

File: drakrun/drakrun/draksetup.py [hunk 5]
```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
...
+
+    # Upload profiles
+    for file in os.listdir(PROFILE_DIR):
+        logging.info("Uploading profile %s", file)
+        mc.fput_object(bucket, f"{name}/profiles/{file}", os.path.join(PROFILE_DIR, file))
+
+
+def do_import_full(mc, name, bucket, zpool):
+    """ Perform full snapshot import, symmetric to do_export_full """
+    do_import_minimal(mc, name, bucket, zpool)
```
Comment (BonusPlay) [inline drakrun/drakrun/draksetup.py:667]: This will print message from end of `do_import_minimal`.

**Review by BonusPlay — CHANGES_REQUESTED** [review]

File: drakrun/drakrun/draksetup.py (hunk 1 above)
Comment (chivay) [reply drakrun/drakrun/draksetup.py:491]: :+1:

File: drakrun/drakrun/draksetup.py (hunk 2 above)
Comment (chivay) [reply drakrun/drakrun/draksetup.py:495]: :+1

File: drakrun/drakrun/draksetup.py (hunk 3 above)
Comment (chivay) [reply drakrun/drakrun/draksetup.py:599]: will add a comment

File: drakrun/drakrun/draksetup.py (hunk 5 above)
Comment (chivay) [reply drakrun/drakrun/draksetup.py:663]: :+1:

**Review by BonusPlay — APPROVED** [review]
Not tested manually, just code review

**Pull request merged**